ui/src/app/pages/workspace/invalid-billing-banner.spec.tsx

```diff
@@ -0,0 +1,139 @@
+import '@testing-library/jest-dom';
+
+import * as React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+
+import { ProfileApi, User } from 'generated/fetch';
+
+import { screen } from '@testing-library/dom';
+import { render } from '@testing-library/react';
+import { InvalidBillingBanner } from 'app/pages/workspace/invalid-billing-banner';
+import { registerApiClient } from 'app/services/swagger-fetch-clients';
+import { plusDays } from 'app/utils/dates';
+import { currentWorkspaceStore } from 'app/utils/navigation';
+import { serverConfigStore } from 'app/utils/stores';
+
+import defaultServerConfig from 'testing/default-server-config';
+import {
+  ProfileApiStub,
+  ProfileStubVariables,
+} from 'testing/stubs/profile-api-stub';
+import { workspaceDataStub } from 'testing/stubs/workspaces';
+
+// Mock ReactDOM.createPortal since ToastBanner uses portals
+jest.mock('react-dom', () => {
+  return {
+    ...jest.requireActual('react-dom'),
+    createPortal: (element) => element,
+  };
+});
+
+describe('InvalidBillingBanner', () => {
+  const warningThresholdDays = 5; // arbitrary
+  const me: User = {
+    userName: ProfileStubVariables.PROFILE_STUB.username,
+    givenName: 'Me',
+    familyName: 'Myself',
+  };
+  const someOneElse: User = {
+    userName: 'someOneElse@fake-research-aou.org',
+    givenName: 'Someone',
+    familyName: 'Else',
+  };
+
+  const component = () => {
+    const workspace = currentWorkspaceStore.getValue();
+    const rendered = render(
+      <MemoryRouter>
+        <InvalidBillingBanner
+          onClose={jest.fn()}
+          workspace={workspace}
+          profile={{
+            ...ProfileStubVariables.PROFILE_STUB,
+            ...me,
+          }}
+        />
+      </MemoryRouter>
+    );
+    return rendered;
+  };
+
+  const expectEditWorkspaceButtonExists = () =>
+    screen.getAllByRole('button', { name: /edit workspace/i });
+
+  const expectEditWorkspaceButtonDoesNotExist = () =>
+    expect(
+      screen.queryByRole('button', { name: /edit workspace/i })
+    ).not.toBeInTheDocument();
+
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('December 13, 2016 14:54:00'));
```

Comment:
Does this relate to any other test values?  Does it need to be in the future?  Or is it arbitrary?